### ISSUE TITLE
unable to download you-demi vids

### ISSUE THREAD

**Issue by sevakmaheshr** [rg3/youtube-dl#12082]
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2017.02.10**

### Before submitting an *issue* I have made sure I have:
- [x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [x] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

Till a few versions back, YTDL worked fine downloading you-demi vids. But, now it is consistently croaking out. Since I update my ytdl on a regular basis (every 2 days) I am NOT sure what version worked. But, my current version (2017.02.10) as of Feb 11, 2017 (IST) does NOT work.

I can provide a verbose output if you need, but the only thing I would suggest is that you please look through what you have changed for you-demi downloading during the past 3-5 versions and (maybe) revert the changes? And if nothing has changed in the you-demi downloader, then please look at what may have changed in your core functionality within the last few days (not even weeks) that downloading you-demi vids is broken.

Also, let me know how I can roll-back to an earlier version of ytdl.

Thanks!

**Comment by dstftw** [comment]
No changes in udemy extractor for last 5 months. Stop posting duplicates.

**Comment by sevakmaheshr** [comment]
as much as I appreciate the fantastically useful s/w you have built, are you at least going to tell us if you are going to fix this or if it is fixed?

**Comment by sevakmaheshr** [comment]
Here is an output of the broken download (url changed):

[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'-c', u'-u', u'PRIVATE', u'-p', u'PRIVATE', u'https://www.you-demi.com/xyz-programming-language/']
[debug] Encodings: locale UTF-8, fs utf-8, out None, pref UTF-8
[debug] youtube-dl version 2017.02.10
[debug] Python version 2.7.10 - Darwin-15.6.0-x86_64-i386-64bit
[debug] exe versions: none
[debug] Proxy map: {}

... The first 10 videos have already been downloaded ... so that just goes through quickly ...

[download] 100% of 92.10MiB
[download] Downloading video 11 of 171
[udemy] 5989572: Downloading webpage
ERROR: Unable to download webpage: <urlopen error [Errno 60] Operation timed out> (caused by URLError(error(60, 'Operation timed out'),))
  File "/usr/bin/youtube-dl/youtube_dl/extractor/common.py", line 416, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2006, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/bin/youtube-dl/youtube_dl/utils.py", line 1048, in https_open
    req, **kwargs)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 1197, in do_open
    raise URLError(err)

**Comment by dstftw** [comment]
As clearly seen from your log the problem is in your network connection.

**Comment by sevakmaheshr** [comment]
thank you dstfw. but, how do I find out this problem (which line) so I can at least bring it up to my ISP.

**Comment by dstftw** [comment]
```
ERROR: Unable to download webpage: <urlopen error [Errno 60] Operation timed out> (caused by URLError(error(60, 'Operation timed out'),))
```
>Operation timed out

**Comment by sevakmaheshr** [comment]
Hey, thanks a lot for that reply. One more thing ... if I use the you-demi app on my tablet ... I am able to download the videos on the same connection. But, on my Mac ... with standard python (2.7.something?) ... youtube-dl croaks after some time. See if that info helps.

I am calling my ISP right now. But, I really have a feeling that something changed in the ytdl code that seems to have broken this. Really sorry for bothering you with this. And I truly, sincerely appreciate the software you have built for us. But, I really need to download this course and it is becoming frustrating that the thing that worked just a week ago is NOT working now. :(

MORE FYI ... if you have time to spare:
* I checked the history on you-demi script and realized that you have NOT updated that since Sep 2016
* I downloaded an older version (31.01.17) and tried using that "python ytdl-31.01.17)" but even that seems to croak.

I am truly, sincerely, really sorry for bothering you with this, but I would really like to see my fave ytdl working on you-demi so I can continue studying.

Your help is MUCH, much appreciated.
